### PR TITLE
Setup circleci test server/CI evironment

### PR DESCRIPTION
**Pull request by mortenskoett**
WIth this PR it should be possible - in a very basic way - to run tests automagically on the test server whenever a push to this repositoiry is made. The pipeline all the way to deployment is almost done however I felt this intermediate step should go to master. 

The server spins up a nodejs and a python docker container and runs the python test suite against the api. (I hope!! If you are not convinced b/c you cannot break a test, please tell someone!)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            # docker run -it --rm --network=minitwit-network $DOCKER_USERNAME/minitwit-tests
+
+    #   - run: 
+    #       name: Build and push to Dockerhub
+    #       command: |
+    #         echo "Login"
+    #         echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+
+    #         echo "Build app"
+    #         docker build -t $DOCKER_USERNAME/minitwit-app . -f Dockerfile-minitwit
```
Comment (livborre) [inline .circleci/config.yml]: missing . at the end?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    #         docker build -t $DOCKER_USERNAME/minitwit-tests -f Dockerfile-minitwit-tests .
+
+    #         echo "Push"
+    #         docker push $DOCKER_USERNAME/minitwit-app
+    #         docker push $DOCKER_USERNAME/minitwit-tests
+
+
+    #         # docker build -t $DOCKER_USERNAME/minitwit-app -f Dockerfile-minitwit .
+    #         # docker build -t $DOCKER_USERNAME/minitwit-tests -f Dockerfile-minitwit-tests .
```
Comment (livborre) [inline .circleci/config.yml]: redundant?

File: .circleci/config.yml (hunk 1 above)
Comment (livborre) [inline .circleci/config.yml]: missing . at the end?

File: dockerfiles/Dockerfile-minitwit-tests [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.7-alpine
+
+WORKDIR /minitwit-api-tests
+
+COPY ./server/tests/* ./
+
+RUN pip install -r requirements.txt
+
+ENTRYPOINT ["pytest", "-s"]
```
Comment (livborre) [inline dockerfiles/Dockerfile-minitwit-tests]: should the test recide in a seperate container?

File: .circleci/config.yml (hunk 2 above)
Comment (mortenskoett) [reply .circleci/config.yml]: Yes most possible. Its very much a WIP. It depends a little on how the script is continued in relation to deployment! Well spotted!

File: .circleci/config.yml (hunk 1 above)
Comment (mortenskoett) [reply .circleci/config.yml]: Bug sniper! Yes, but atm it is commented out as a WIP in relation to deployment.

File: dockerfiles/Dockerfile-minitwit-tests (hunk 3 above)
Comment (mortenskoett) [reply dockerfiles/Dockerfile-minitwit-tests]: Not necessarily I guess. This was done because the tests need a python environment.

**Comment by mortenskoett** [comment]
Alright. I will just clean up some of this and merge it with master and work from master. Hang on...

**Pull request closed without merging**